main.cc

```diff
@@ -433,6 +433,39 @@ static int scylla_main(int ac, char** av) {
         exit(1);
     }
 
+    // Even on the environment which causes error during initalize Scylla,
+    // "scylla --version" should be able to run without error.
+    // To do so, we need to parse and execute these options before
+    // initializing Scylla/Seastar classes.
+    bpo::options_description preinit_description("Scylla options");
+    bpo::variables_map preinit_vm;
+    preinit_description.add_options()
+        ("version", bpo::bool_switch(), "print version number and exit")
+        ("build-id", bpo::bool_switch(), "print build-id and exit")
+        ("build-mode", bpo::bool_switch(), "print build mode and exit")
+        ("list-tools", bpo::bool_switch(), "list included tools and exit");
```

Comment:
I think you can drop these from the other options parser, since we'll never reach that if one of these options is specificed.

Reply:
Well, yes, it won't reach to original option handler defined for main options parser, but why I didn't drop them was because to keep current --help message.
If we simply drop them, description about these options will disappear from --help message.
We can show it again by adding `preinit_description` to main parser like this:
```
    app.get_options_description().add(preinit_description);
```

But then, --help looks different than original one since preinit_description shows as individual option group, like this:
```
parsed command line options: [help]
scylla - NoSQL data store using the seastar framework

For more information, see https://github.com/scylladb/scylla.

The scylla executable hosts tools in addition to the main scylla server, these
can be invoked as: scylla {tool_name} [...]

For a list of available tools, run: scylla --list-tools
For more information about individual tools, run: scylla {tool_name} --help

To start the scylla server proper, simply invoke as: scylla server (or just scylla).

Scylla options:
  -h [ --help ]                         show help message
  --help-seastar                        show help message about seastar options
  --help-loggers                        print a list of logger names and exit
  --options-file arg                    configuration file (i.e. 
                                        <SCYLLA_HOME>/conf/scylla.yaml)
... skipped lines ...
  --force-schema-commit-log arg         Use separate schema commit log 
                                        unconditionally rater than after 
                                        restart following discovery of 
                                        cluster-wide support for it.

Scylla version options:
  --version                             print version number and exit
  --build-id                            print build-id and exit
  --build-mode                          print build mode and exit
  --list-tools                          list included tools and exit

Deprecated options - ignored:
  --background-writer-scheduling-quota arg
  --auto-adjust-flush-quota arg
```

Reply:
Ok.